include/wx/combo.h

```diff
@@ -585,6 +585,18 @@ class WXDLLIMPEXP_CORE wxComboCtrlBase : public wxControl,
 
     void OnPopupMouseEvent(wxMouseEvent& event);
 
+    // This function can be used as event handle for wxEVT_DPI_CHANGED event
+    // and simply recalculates button size when it happens.
+    void WXHandleDPIChanged(wxDPIChangedEvent& event)
+    {
+        // Drop button size
+        m_btnSize = wxSize(1, 1);
```

Comment:
This doesn't work, we have
```cpp
    if ( m_btnSize.x > 0 )
        return m_btnSize;
```
in `wxComboCtrlBase::GetButtonSize()`, so you really need to
```suggestion
        // Ensure it is really recalculated.
        m_btnSize = wxDefaultSize;
```
(using `wxSize()` would work too, but not `1, 1` as you do).